File: src/0167-两数之和II/0167.cpp
// 双指针法
class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        int left = 0, right = numbers.size() - 1;
        int sum;
        while (left < right)
        {
            sum = numbers[left] + numbers[right];
            if (sum == target)
                break;
            else if (sum < target)
                left++;
            else
                right--;
        }
        return vector<int>{left + 1, right + 1};
    }
};

File: src/0167-两数之和II/0167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        num_dict = {}
        for i,item in enumerate(numbers):
            last = target - item
            if last in num_dict:
                return [num_dict[last]+1,i+1]
            else:
                num_dict[item] = i

        return []